account_bank_statement_import_online_qonto/models/online_bank_statement_provider_qonto.py

```diff
@@ -0,0 +1,122 @@
+        self.ensure_one()
+        iban = sanitize_account_number(journal.bank_account_id.acc_number)
+        url = QONTO_ENDPOINT + '/transactions'
+        params = {'slug': slug, 'iban': iban}
+        if date_since:
+            params['settled_at_from'] = date_since.replace(microsecond=0).isoformat() + 'Z'
+        if date_until:
+            params['settled_at_to'] = date_until.replace(microsecond=0).isoformat() + 'Z'
+        response = requests.get(url, verify=False, params=params, headers=self._qonto_header())
+        if response.status_code == 200:
+            return json.loads(response.text)
+        raise UserError(_('%s \n\n %s') % (response.status_code, response.text))
+
+    def _qonto_obtain_statement_data(self, date_since, date_until):
+        self.ensure_one()
+        BankStatement = self.env['account.bank.statement']
+        BankStatementLine = self.env['account.bank.statement.line']
+
+        slugs = self._qonto_get_slug()
+
+        journal = self.journal_id
+
+        slug = slugs.get(sanitize_account_number(journal.bank_account_id.acc_number))
+        if not slug:
+            raise UserError(
+                _('Qonto : wrong configuration, unknow account %s') % journal.bank_account_id.acc_number)
+
+        data = self._qonto_get_transaction(journal, slug, date_since, date_until)
+
+        currency = journal.currency_id or journal.company_id.currency_id
+
+        new_transactions = []
+        sequence = 0
+        for transaction in data.get('transactions', []):
+            if BankStatementLine.sudo().search(
+                [('unique_import_id', '=', transaction['transaction_id']),
+                 ('journal_id', '=', journal.id)], limit=1):
+                continue
+            date = datetime.strptime(transaction['settled_at'], '%Y-%m-%dT%H:%M:%S.%fZ')
+            date = fields.Date.to_date(date.astimezone(pytz.timezone('Europe/Paris')))
+            side = 1 if transaction['side'] == 'credit' else -1
+            sequence += 1
+            vals_line = {
+                'sequence': sequence,
+                'date': date,
+                'name': re.sub(' +', ' ', '%s %s' % (transaction['label'], transaction['reference'])) or '/',
+                'ref': transaction['reference'],
+                'unique_import_id': transaction['transaction_id'],
+                'amount': transaction['amount'] * side,
+            }
+
+            line_currency = self.env['res.currency'].search(
+                [('name', '=', transaction['local_currency'])], limit=1)
+
+            if currency != line_currency:
+                vals_line.update({'currency_id': currency.id,
+                                  'amount_currency': transaction['local_amount'] * side})
+
+            new_transactions.append(vals_line)
+        if new_transactions:
+            date = max([t['date'] for t in new_transactions])
+            start_balance = BankStatement.search([('journal_id', '=', journal.id)],
```

Comment:
This is cheating, if provider does not provide the balance, leave it out

Reply:
On top of that, that's done by base module in consistent way

Reply:
the provider doesn't provide the balance at date, juste the today balance.

Reply:
Then don't mimic that provider gives the balance. Odoo will take previous statement balance by default, and providing balance_end_real is simply wrong, since you don't have real balance.